### PR TITLE
fix(.goreleaser.yml): Load version for wasmvm dynamically

### PR DESCRIPTION
**Pull request by Unique-Divine**
# Purpose

The changes here should make `goreleaser` load whichever version is stated in our `go.mod`. 

- See brief discussion: https://github.com/NibiruChain/nibiru/pull/1629#issuecomment-1753764818

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -9,7 +9,8 @@ builds:
     binary: nibid
     hooks:
```
Comment (Unique-Divine) [inline .goreleaser.yml:10]: Just a note, I believe setting the `$WASMVM_VERSION` in the `env` block won't be seen by the environment in the `pre` blocks. This should make `goreleaser` load whichever version is stated in our `go.mod`.

**Comment by Unique-Divine** [comment]
@NibiruChain/backend , how should I test this? Just tag a prerelease and then delete it?

**Comment by helder-moreira** [comment]
> @NibiruChain/backend , how should I test this? Just tag a prerelease and then delete it?  

You can test locally with `make release-snapshot`.

**Comment by Unique-Divine** [comment]
> > @NibiruChain/backend , how should I test this? Just tag a prerelease and then delete it?
> 
> You can test locally with `make release-snapshot`.

### That worked for me. I got the following output

  • starting release...
  • loading config file                              file=.goreleaser.yml
  • loading environment variables
  • getting and validating git state
    • building...                                    commit=ef307eb3a73ad887c3a8660c55092b0f7c41153f latest tag=v0.22.0-alpha.1
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
  • snapshotting
    • building snapshot...                           version=0.22.0-alpha.1-SNAPSHOT-ef307eb3
  • checking distribution directory
    • --rm-dist is set, cleaning it up
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/linux_linux_arm64/nibid
    • building                                       binary=dist/darwin_darwin_amd64_v1/nibid
    • running hook                                   hook=wget https://github.com/CosmWasm/wasmvm/releases/download/v1.4.0/libwasmvm_muslc.x86_64.a -O /usr/lib/x86_64-linux-gnu/libwasmvm_muslc.a
    • building                                       binary=dist/darwin_darwin_arm64/nibid
    • building                                       binary=dist/linux_linux_amd64_v1/nibid
    • running hook                                   hook=wget https://github.com/CosmWasm/wasmvm/releases/download/v1.4.0/libwasmvmstatic_darwin.a -O /osxcross/target/SDK/MacOSX12.0.sdk/usr/lib/libwasmvmstatic_darwin.a
    • running hook                                   hook=wget https://github.com/CosmWasm/wasmvm/releases/download/v1.4.0/libwasmvmstatic_darwin.a -O /osxcross/target/SDK/MacOSX12.0.sdk/usr/lib/libwasmvmstatic_darwin.a
    • running hook                                   hook=wget https://github.com/CosmWasm/wasmvm/releases/download/v1.4.0/libwasmvm_muslc.x86_64.a -O /usr/lib/x86_64-linux-gnu/libwasmvm_muslc.a
    • running hook                                   hook=wget https://github.com/CosmWasm/wasmvm/releases/download/v1.4.0/libwasmvm_muslc.aarch64.a -O /usr/lib/aarch64-linux-gnu/libwasmvm_muslc.a
    • running hook                                   hook=wget https://github.com/CosmWasm/wasmvm/releases/download/v1.4.0/libwasmvm_muslc.aarch64.a -O /usr/lib/aarch64-linux-gnu/libwasmvm_muslc.a
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/include.8.1.1.tar.gz -O /root/include.8.1.1.tar.gz
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/include.8.1.1.tar.gz -O /root/include.8.1.1.tar.gz
    • running hook                                   hook=tar -xf /root/include.8.1.1.tar.gz -C /osxcross/target/SDK/MacOSX12.0.sdk/usr/include/
    • running hook                                   hook=tar -xf /root/include.8.1.1.tar.gz -C /osxcross/target/SDK/MacOSX12.0.sdk/usr/include/
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/librocksdb_8.1.1_darwin_all.tar.gz -O /root/librocksdb_8.1.1_darwin_all.tar.gz
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/librocksdb_8.1.1_darwin_all.tar.gz -O /root/librocksdb_8.1.1_darwin_all.tar.gz
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/include.8.1.1.tar.gz -O /root/include.8.1.1.tar.gz
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/include.8.1.1.tar.gz -O /root/include.8.1.1.tar.gz
    • running hook                                   hook=tar -xvf /root/include.8.1.1.tar.gz -C /usr/include/
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/librocksdb_8.1.1_linux_amd64.tar.gz -O /root/librocksdb_8.1.1_linux_amd64.tar.gz
    • running hook                                   hook=tar -xvf /root/include.8.1.1.tar.gz -C /usr/include/
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/librocksdb_8.1.1_linux_amd64.tar.gz -O /root/librocksdb_8.1.1_linux_amd64.tar.gz
    • running hook                                   hook=tar -xf /root/librocksdb_8.1.1_darwin_all.tar.gz -C /osxcross/target/SDK/MacOSX12.0.sdk/usr/lib/
    • running hook                                   hook=tar -xf /root/librocksdb_8.1.1_darwin_all.tar.gz -C /osxcross/target/SDK/MacOSX12.0.sdk/usr/lib/
    • running hook                                   hook=tar -xvf /root/librocksdb_8.1.1_linux_amd64.tar.gz -C /usr/lib/x86_64-linux-gnu/
    • running hook                                   hook=tar -xvf /root/librocksdb_8.1.1_linux_amd64.tar.gz -C /usr/lib/x86_64-linux-gnu/
    • overridden build details for darwinarm64       details={ [-s -w -X github.com/cosmos/cosmos-sdk/version.Name=nibiru -X github.com/cosmos/cosmos-sdk/version.AppName=nibid -X github.com/cosmos/cosmos-sdk/version.Version={{ .Version }} -X github.com/cosmos/cosmos-sdk/version.Commit={{ .Commit }} -X github.com/cosmos/cosmos-sdk/types.DBBackend=rocksdb -linkmode=external -extldflags '-lrocksdb -lstdc++ -ldl -lz -lbz2'] [netgo osusergo ledger static static_wasm rocksdb grocksdb_no_link] [-mod=readonly -trimpath] [] [] [CC=oa64-clang]}
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/librocksdb_8.1.1_linux_arm64.tar.gz -O /root/librocksdb_8.1.1_linux_arm64.tar.gz
    • running hook                                   hook=wget https://github.com/NibiruChain/gorocksdb/releases/download/v8.1.1/librocksdb_8.1.1_linux_arm64.tar.gz -O /root/librocksdb_8.1.1_linux_arm64.tar.gz
    • running hook                                   hook=tar -xvf /root/librocksdb_8.1.1_linux_arm64.tar.gz -C /usr/lib/aarch64-linux-gnu/
    • running hook                                   hook=tar -xvf /root/librocksdb_8.1.1_linux_arm64.tar.gz -C /usr/lib/aarch64-linux-gnu/
    • overridden build details for linuxarm64        details={ [-s -w -X github.com/cosmos/cosmos-sdk/version.Name=nibiru -X github.com/cosmos/cosmos-sdk/version.AppName=nibid -X github.com/cosmos/cosmos-sdk/version.Version={{ .Version }} -X github.com/cosmos/cosmos-sdk/version.Commit={{ .Commit }} -X github.com/cosmos/cosmos-sdk/types.DBBackend=rocksdb -linkmode=external -extldflags '-static -lrocksdb -lstdc++ -lm -ldl -lpthread'] [netgo osusergo ledger muslc rocksdb grocksdb_no_link static] [-mod=readonly -trimpath] [] [] [CC=aarch64-linux-gnu-gcc]}
    • took: 11m6s
  • universal binaries
    • creating from 2 binaries                       binary=dist/darwin_darwin_all/nibid id=darwin
  • archives
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-ef307eb3_darwin_all.tar.gz
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-ef307eb3_darwin_amd64.tar.gz
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-ef307eb3_linux_amd64.tar.gz
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-ef307eb3_darwin_arm64.tar.gz
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-ef307eb3_linux_arm64.tar.gz
    • took: 29s
  • calculating checksums
  • storing release metadata
    • writing                                        file=dist/artifacts.json
    • writing                                        file=dist/metadata.json
  • release succeeded after 11m36s

**Comment by Unique-Divine** [comment]
Just confirmed that this works locally if you use scripts in the `hooks.pre` block instead of setting an environment variable.

### Output from commit 2fc1f0318df3c498f9d829da90947004189ce1c1

```bash
make release-snapshot
```


docker run \
        --rm \
        --platform linux/amd64 \
        -v "/home/realu/warpath/NibiruChain/nibiru":/go/src/github.com/NibiruChain/nibiru \
        -w /go/src/github.com/NibiruChain/nibiru \
        -e CGO_ENABLED=1 \
        -e GITHUB_TOKEN= \
        goreleaser/goreleaser-cross:v1.19.4 \
        release --rm-dist --snapshot
  • starting release...
  • loading config file                              file=.goreleaser.yml
  • loading environment variables
  • getting and validating git state
    • building...                                    commit=365014d0d395dceed51982c11d474a23b997ef61 latest tag=v0.22.0-alpha.1
    • pipe skipped                                   reason=disabled during snapshot mode
    • took: 1s
  • parsing tag
  • setting defaults
  • snapshotting
    • building snapshot...                           version=0.22.0-alpha.1-SNAPSHOT-365014d0
  • checking distribution directory
    • --rm-dist is set, cleaning it up
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/darwin_darwin_arm64/nibid
    • building                                       binary=dist/darwin_darwin_amd64_v1/nibid
    • building                                       binary=dist/linux_linux_amd64_v1/nibid
    • building                                       binary=dist/linux_linux_arm64/nibid
    • running hook                                   hook=bash contrib/scripts/release_pre_darwin.sh
    • running hook                                   hook=bash contrib/scripts/release_pre_linux.sh
    • running hook                                   hook=bash contrib/scripts/release_pre_darwin.sh
    • running hook                                   hook=bash contrib/scripts/release_pre_linux.sh
    • overridden build details for darwinarm64       details={ [-s -w -X github.com/cosmos/cosmos-sdk/version.Name=nibiru -X github.com/cosmos/cosmos-sdk/version.AppName=nibid -X github.com/cosmos/cosmos-sdk/version.Version={{ .Version }} -X github.com/cosmos/cosmos-sdk/version.Commit={{ .Commit }} -X github.com/cosmos/cosmos-sdk/types.DBBackend=rocksdb -linkmode=external -extldflags '-lrocksdb -lstdc++ -ldl -lz -lbz2'] [netgo osusergo ledger static static_wasm rocksdb grocksdb_no_link] [-mod=readonly -trimpath] [] [] [CC=oa64-clang]}
    • overridden build details for linuxarm64        details={ [-s -w -X github.com/cosmos/cosmos-sdk/version.Name=nibiru -X github.com/cosmos/cosmos-sdk/version.AppName=nibid -X github.com/cosmos/cosmos-sdk/version.Version={{ .Version }} -X github.com/cosmos/cosmos-sdk/version.Commit={{ .Commit }} -X github.com/cosmos/cosmos-sdk/types.DBBackend=rocksdb -linkmode=external -extldflags '-static -lrocksdb -lstdc++ -lm -ldl -lpthread'] [netgo osusergo ledger muslc rocksdb grocksdb_no_link static] [-mod=readonly -trimpath] [] [] [CC=aarch64-linux-gnu-gcc]}
    • took: 8m34s
  • universal binaries
    • creating from 2 binaries                       binary=dist/darwin_darwin_all/nibid id=darwin
  • archives
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-365014d0_darwin_arm64.tar.gz
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-365014d0_linux_arm64.tar.gz
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-365014d0_darwin_all.tar.gz
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-365014d0_darwin_amd64.tar.gz
    • creating                                       archive=dist/nibid_0.22.0-alpha.1-SNAPSHOT-365014d0_linux_amd64.tar.gz
    • took: 32s
  • calculating checksums
  • storing release metadata
    • writing                                        file=dist/artifacts.json
    • writing                                        file=dist/metadata.json
  • release succeeded after 9m7s

**Pull request merged**